src/com/google/enterprise/adaptor/ValidatedUri.java

```diff
@@ -35,49 +37,74 @@
   /** The connect timeout, in milliseconds. */
   private static final int TIMEOUT_MILLIS = (int) TimeUnit.SECONDS.toMillis(30);
 
+  /** The validated URI. */
+  private final URI uri;
+
   /**
-   * Attempts to validate the given URL syntax and host reachability.
-   * In this case, we're mostly trying to catch typos.
-   * No attempts to fetch the URL are made, so a URL build from start paths or
-   * using placeholder values in format substitutions may be supplied.
-   * This is intended to be called from the adaptor's init() method, where
-   * InvalidConfigurationExceptions are handled gracefully.
+   * Attempts to validate the given URI syntax with some more stringent checks
+   * than new URI. In this case, we're mostly trying to catch typos and obvious
+   * configuration issues.
    *
-   * @param urlString the URL to test
-   * @return {@code true} if the URL's host is reachable, {@code false}
-   *         if the host is not reachable.
-   * @throws InvalidConfigurationException if the URL syntax is invalid
+   * @param uriString the URI to test
+   * @throws URISyntaxException if the URL syntax is invalid
    */
-  public boolean validate(String urlString) throws InvalidConfigurationException {
-    URL url;
+  public ValidatedUri(String uriString) throws URISyntaxException {
+    if (uriString == null) {
+      throw new URISyntaxException("null", "null URI");
+    }
     try {
-      // basic syntax checking, with more understandable error messages
-      url = new URL(urlString);
-      // advanced syntax checking, with more cryptic error messages
-      url = new URI(urlString).toURL();
+      // Basic syntax checking, with more understandable error messages.
+      // Also ensures the URI is a URL, not a URN.
+      URL url = new URL(uriString);
+      // Advanced syntax checking, with more cryptic error messages.
+      uri = new URI(uriString);
+      url = uri.toURL();
```

Comment:
Does toURL() have a side-effect? Maybe don't bother assigning it back to url if it does?